### PR TITLE
Env var warnings and readme for ETL tasks/commands

### LINKED ISSUE #175: Document ETL commands & required settings
# Description/Context
In order for developers to work on MIT Open, they need test data—courses, videos, programs, etc. Test data can be acquired by running our ETL pipelines. 

However, the required settings are not very well documented: For example, if `XPRO_CATALOG_API_URL` and `XPRO_COURSES_API_URL` are both unset, then `./manage.py backpopulate_xpro_data` will run successfully—no errors thrown or in logs—but course/program data is not retrieved.

We should document our ETL pipelines and their required environment variables.

# Plan/Design
1. Our readme should include documentation about the ETL pieplines. (IMO, not much is needed. E.g., "See `./manage.py backpopulate_xpro_data --help` and other `backpopulate_xyz_commands`" would be fine.)
2. The backpopulate commands should throw easily discoverable errors if they do not function as expected. (E.g., env vars are missing).

### PR DESCRIPTION
**Pull request by ChristopherChudzicki**
# What are the relevant tickets?
Closes #175 

# Description (What does it do?)
This PR:
- Adds more warnings for missing env vars in many of the ETL pipelines. *Some had them, some didn't.*
- Adds a line similar to `"Populated {count} resources. See celery logs for details."` to most of the ETL backpopulate management commands.
- Updates the README a bit

# How can this be tested?
I have tested this by running the following management commands, with and without relevant environment variables, which you can find in `open_discussions/settings_course_etl.py`.
```
docker compose exec web ./manage.py backpopulate_micromasters_data
docker compose exec web ./manage.py backpopulate_mit_edx_data
docker compose exec web ./manage.py backpopulate_mitxonline_data
docker compose exec web ./manage.py backpopulate_ocw_data
docker compose exec web ./manage.py backpopulate_oll_data
docker compose exec web ./manage.py backpopulate_prolearn_data
docker compose exec web ./manage.py backpopulate_xpro_data
```

### CODE REVIEW

File: learning_resources/etl/micromasters.py [hunk 1]
```diff
@@ -20,6 +24,7 @@ def extract():
             headers={**COMMON_HEADERS},
             timeout=settings.REQUESTS_TIMEOUT,
         ).json()
+    log.warning("Missing required setting MICROMASTERS_CATALOG_API_URL")
```
Comment (ChristopherChudzicki) [inline learning_resources/etl/micromasters.py:27]: Some of the ETL tasks issue warnings for missing env vars already, others do not.

My understanding for why these are warnings not errors is that we want the celery tasks (which are run regularly in local dev and on rc/prod, and via management commands) to be no ops when the env vars are missing.

File: README.md [hunk 2]
```diff
@@ -134,32 +149,18 @@ otherwise do in a Django shell. To get started:
...
-  Here is a one-line command that will produce a browser-ready URL from that output. Run this in a separate terminal:
-  ```bash
-  APP_HOST="open.ol.local"; docker logs $(docker ps --format '{{.Names}}' | grep "_notebook_run_") | grep -E "http://(.*):8080[^ ]+\w" | tail -1 | sed -e 's/^[[:space:]]*//' | sed -e "s/(.*)/$APP_HOST/"
+  docker compose -f docker-compose-notebook.yml run --rm --service-ports notebook
   ```
-  OSX users can pipe that output to `xargs open` to open a browser window directly with the URL from that command.
-- Click the `.ipynb` file that you created to run the notebook
-- Execute the first block to confirm it's working properly (click inside the block
-  and press Shift+Enter)
+- The notebook container log output will indicate a URL at which you can interact with the notebook server.
```
Comment (ChristopherChudzicki) [inline README.md:159]: The url seems to be directly usable.

File: learning_resources/etl/oll.py [hunk 3]
```diff
@@ -8,17 +10,36 @@
...
         settings.OLL_API_CLIENT_SECRET,
         settings.OLL_API_ACCESS_TOKEN_URL,
         settings.OLL_API_URL,
         settings.OLL_BASE_URL,
         settings.OLL_ALT_URL,
-        PlatformType.oll.name,
+        PlatformType.edx.name,
         OfferedBy.mitx.name,
-        ETLSource.oll.name,
+        ETLSource.mit_edx.name,
```
Comment (mbertrand) [inline learning_resources/etl/oll.py:40]: The original values are the correct ones (`PlatformType.oll.name`, `ETLSource.oll.name,`)

**Review by mbertrand — CHANGES_REQUESTED** [review]
looks good except for one thing

File: learning_resources/etl/oll.py (hunk 3 above)
Comment (ChristopherChudzicki) [reply learning_resources/etl/oll.py:40]: Ack, thanks for that catch.

**Review by mbertrand — APPROVED** [review]
👍

**Pull request merged**